### PR TITLE
Automate firmware

### PR DESCRIPTION
**Pull request by sharpenedblade**
This attempts to get the firmware from the macos apfs partition. If it fails for any reason, boot continues normally.

### CODE REVIEW

File: docs/tools/firmware.sh [hunk 1]
```diff
@@ -832,11 +845,11 @@ case "$os" in
 				echo "Copying this script to EFI"
 				cp "$0" "/Volumes/${EFILABEL}/firmware.sh" 2>/dev/null \
 					|| curl -s https://wiki.t2linux.org/tools/firmware.sh > "/Volumes/${EFILABEL}/firmware.sh" \
-					|| (echo && cat <<- EOF && read -r
+					|| (
+						echo && cat <<- EOF && [ "$interactive" = "true"] && echo "Press enter after you have copied" && read -r
```
Comment (AdityaGarg8) [inline docs/tools/firmware.sh:849]: Shouldn't there be a blank line before Press enter after you have copied?

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

File: docs/tools/firmware.sh (hunk 1 above)
Comment (sharpenedblade) [reply docs/tools/firmware.sh:849]: It looks exactly the same as before, but I can add a extra newline if you want.

Comment (AdityaGarg8) [reply docs/tools/firmware.sh:849]: > It looks exactly the same as before, but I can add a extra newline if you want.

It's fine then

**Comment by AdityaGarg8** [comment]
You've repeated "Detected Linux" and "Detected macOS". You probably want to remove them.

**Comment by sharpenedblade** [comment]
> You've repeated "Detected Linux" and "Detected macOS". You probably want to remove them.
> 
> L908 and L816

Fixed

**Comment by AdityaGarg8** [comment]
There are still some functions that you have not excluded in no interactive. Eg: the one asking to keep firmware in Efi or not when option 1 is chosen on Linux. Can you review the whole script?

**Comment by sharpenedblade** [comment]
Fixed, thats the last `read`

**Comment by AdityaGarg8** [comment]
image

**Comment by AdityaGarg8** [comment]
image

**Comment by AdityaGarg8** [comment]
Please do proper testing of the script.

**Comment by sharpenedblade** [comment]
> Please do proper testing of the script.

It should work now. I only tested it on linux though.

**Comment by AdityaGarg8** [comment]
> image

This one still doesn't work

> image

This one still doesn't work

**Comment by AdityaGarg8** [comment]
Also you may add your Copyright at the top if you want.

**Pull request merged**